### PR TITLE
sets: add the Add() method

### PR DESCRIPTION
**Pull request by marco-m-pix4d**
Part of PCI-3790

### CODE REVIEW

**Review by aliculPix4D — APPROVED** [review]

File: sets/sets.go [hunk 1]
```diff
@@ -60,6 +60,15 @@ func (s *Set[T]) Contains(item T) bool {
 	return found
 }
 
+// Add inserts item into s. Returns true if the item was present.
+func (s *Set[T]) Add(item T) bool {
```
[the remark this answers is not in the document]
Comment (marco-m-pix4d) [reply sets/sets.go:64]: Yes, when I realized that Add() was missing, I realized that SkyGuide must be collecting a list and calling sets.From(), which doesn't make sense I think.

[the remark this answers is not in the document]
Comment (aliculPix4D) [reply sets/sets.go:64]: Its not yet used in SkyGuide (I think?). I am using Skyguide Concourse lib in my custom tool to inspect pipelines. There I required set, and there I was using sets.From() sadly...

**Comment by aliculPix4D** [comment]
Also, I think we should start thinking about moving some of this small pkgs to its own repo away from cogito..

e.g. sets, retry etc...

**Comment by iAmoric** [comment]
> Also, I think we should start thinking about moving some of this small pkgs to its own repo away from cogito..
 
Ah, I am tempted to agree with this. I was already confused in the last PR. « ok we use this in fly_helper but why it’s in Cogito ? 🤔 »

**Review by iAmoric — APPROVED** [review]

**Review by SpectreVert — APPROVED** [review]

**Comment by marco-m-pix4d** [comment]
To be 100% clear:

we do not depend on a cogito release; we simply depend on a `go get github.com/Pix4D/cogito@master`, which is clear and 100% pinned (`@master` is mapped to the current git SHA).

I agree that it seems that we are almost ready to take the utility functions into their own (public) repo; when we begun, we didn't know how much reuse we would have had...

**Pull request merged**